tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1190,11 +1221,13 @@ public:
    void Finalize()
    {
       auto fileWritten = false;
-      for (auto &file : fOutputFiles) {
-         if (file) {
+      for (auto &fileStack : fOutputFiles) {
+         while(!fileStack.empty()) {
+            auto &file = fileStack.top();
             file->Write();
             file->Close();
             fileWritten = true;
+            fileStack.pop();
```

Comment:
the original code had a `if (file)` which is not replicated by the new logic. I don't know in what case the file pointers could be null at finalization time, but I'd keep the check just in case...